frame/nomination-pools/src/lib.rs

```diff
@@ -768,6 +768,22 @@ impl<T: Config> BondedPool<T> {
 		Ok(())
 	}
 
+	fn ok_to_rebond(
+		&self,
+		member: &PoolMember<T>,
+		new_funds: BalanceOf<T>,
+	) -> Result<(), DispatchError> {
+		// The member may just be kicked, thus reject to rebond. Note that this check also reject
+		// member who didn't be kicked but happen to fully unbond and try to rebond when the
+		// pool is blocked
+		// TODO: shoule also add this check to `bond_extra`?
+		if matches!(self.state, PoolState::Blocked) && member.active_points().is_zero() {
+			return Err(Error::<T>::FullyUnbonding.into())
+		}
```

Comment:
Intuitively, I think this check should be added in case a member is kicked but utilize `rebond` to un-kick itself.

Reply:
I found `do_reward_payout` will return `FullyUnbonding` error for the member who has fully unbonded, so `bond_extra` will return `FullyUnbonding` error for the member who had been kicked and thus may not need this check (it also means fully unbonded member can't successfully call `bond_extra` again).